events/models.py

```diff
@@ -488,10 +459,11 @@ def save(self, *args, **kwargs):
         super(Event, self).save(*args, **kwargs)
 
         # needed to cache location event numbers
-        if self.location:
-            recache_n_events_in_location(self.location)
+        if not old_location and self.location:
+            Place.objects.filter(id=self.location.id).update(n_events_changed=True)
         if old_location and old_location != self.location:
-            recache_n_events_in_location(old_location)
+            Place.objects.filter(id=old_location.id).update(n_events_changed=True)
```

Comment:
`Place.objects.filter(id__in=(old_location.id, self.location.id))...` to only use 1 query.